### PR TITLE
SlotIndex and EpochIndex as u32

### LINKED ISSUE #1316: Make SlotIndex and EpochIndex u32
(will update when there is a TIP change)

### PR DESCRIPTION
**Pull request by thibault-martinez**
Closes https://github.com/iotaledger/iota-sdk/issues/1316

### CODE REVIEW

**Review by DaughterOfMars — COMMENTED** [review]
good stuff

File: sdk/src/types/block/protocol.rs [hunk 1]
```diff
@@ -163,8 +157,8 @@ impl ProtocolParameters {
     }
 
     /// Returns the slots per epoch of the [`ProtocolParameters`].
-    pub fn slots_per_epoch(&self) -> u64 {
-        2_u64.pow(self.slots_per_epoch_exponent() as u32)
+    pub fn slots_per_epoch(&self) -> u32 {
+        2_u32.pow(self.slots_per_epoch_exponent() as u32)
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs:161]: Could do this
```suggestion
        1 << self.slots_per_epoch_exponent()
```

Comment (thibault-martinez) [reply sdk/src/types/block/protocol.rs:161]: I'd rather keep the pow version tbh, intent is clearer

File: sdk/src/types/block/block_id.rs [hunk 2]
```diff
@@ -34,7 +34,7 @@ pub struct BlockId {
 
 impl BlockId {
     /// The length of a [`BlockId`]
-    pub const LENGTH: usize = 40;
+    pub const LENGTH: usize = 36;
```
Comment (Thoralf-M) [inline sdk/src/types/block/block_id.rs:37]: Since when is this the case?

Comment (thibault-martinez) [reply sdk/src/types/block/block_id.rs:37]: Because slot index is part of the BlockId and it's now 4 bytes shorter

Comment (Thoralf-M) [reply sdk/src/types/block/block_id.rs:37]: Ah sure makes sense, wen TIP updated

Comment (thibault-martinez) [reply sdk/src/types/block/block_id.rs:37]: 🤷🏻

**Review by Thoralf-M — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

File: sdk/src/types/block/protocol.rs (hunk 1 above)
Comment (DaughterOfMars) [reply sdk/src/types/block/protocol.rs:161]: This is how they do it in iota.go

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**